src/Core/Domain/Product/QueryResult/ProductCustomizability.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\QueryResult;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\ProductCustomizabilitySettings;
+
+/**
+ * Transfers data of product customizability value
+ */
+class ProductCustomizability
+{
+    /**
+     * @var int
+     */
+    private $value;
```

Comment:
`$type` ? it's a sort of type, right ?

Reply:
its product->customizable value.
Although Im not sure if we want to call it a type or an option or whatever.
This is the value mentioned here: https://github.com/PrestaShop/PrestaShop/pull/19964#discussion_r452177459
```
as @jolelievre found out, the Product->customizable can have 3 possible values as it seems.

no custom fields (uncostomizable) = 0
at least one custom field (no required) = 1
at least one custom field, and at least one required = 2

Reply:
In this context I feel like "value" is like "data" 😅